src/tiled/editableobject.cpp

```diff
@@ -71,6 +73,63 @@ void EditableObject::removeProperty(const QString &name)
         mObject->removeProperty(name);
 }
 
+QVariantMap EditableObject::propertiesIncludingInherited() const
+{
+    auto mergeVariantMaps = [](QVariantMap& dstMap, const QVariantMap& srcMap) {
+        for (auto iter = srcMap.constBegin(); iter != srcMap.constEnd(); ++iter)
+        {
+            dstMap.insert(iter.key(), iter.value());
+        }
+    };
+
+    QVariantMap allProperties;
+    // Insert properties into allProperties in the reverse order that
+    // Object::inheritedProperty searches them, to make sure that the
+    // same precedence is maintained.
+
+    QString objectType;
+    switch (mObject->typeId()) {
+    case Object::MapObjectType:
+    {
+        auto mapObject = static_cast<const MapObject*>(mObject);
+        objectType = mapObject->type();
+        if (objectType.isEmpty())
+            if (const Tile *tile = mapObject->cell().tile())
+                objectType = tile->type();
+        break;
+    }
+    case Object::TileType:
+        objectType = static_cast<const Tile*>(mObject)->type();
+        break;
+    default:
+        break;
+    }
+
+    if (!objectType.isEmpty()) {
+        for (const ObjectType &type : qAsConst(Object::objectTypes())) {
```

Comment:
`qAsConst` is not needed here, since `Object::objectTypes` already returns `const &`.